services/src/main/java/org/keycloak/keys/loader/PublicKeyStorageManager.java

```diff
@@ -53,4 +64,12 @@ public static PublicKey getIdentityProviderPublicKey(KeycloakSession session, Re
         return keyStorage.getPublicKey(modelKey, kid, loader);
     }
 
+    protected static PublicKey getSavedPublicKey(OIDCIdentityProviderConfig config) {
+        if (config.getPublicKeySignatureVerifier() != null && !config.getPublicKeySignatureVerifier().trim().equals("")) {
+            return PemUtils.decodePublicKey(config.getPublicKeySignatureVerifier());
```

Comment:
PublicKeyStorageManager isn't cached afaik, so this will be pretty bad for performance as decoding pems is rather expensive.